### PR TITLE
Add graph library.

### PR DESCRIPTION
**Pull request by siarheishut**
This library allows to compute the shortest paths in the graph using Floyd-Warshall algorithm.

### CODE REVIEW

File: lib/graph/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.24)
+set(CMAKE_CXX_STANDARD 17)
+project(Graph)
+
+add_library(graph INTERFACE)
+target_sources(graph PUBLIC
+        FILE_SET HEADERS
+        BASE_DIRS src
+        FILES src/graph.h src/router.h)
```
Comment (foxfrikses) [inline lib/graph/CMakeLists.txt:9]: Add empty line.

File: lib/graph/src/graph.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+template<typename Weight>
+typename DirectedWeightedGraph<Weight>::IncidentEdgesRange
+DirectedWeightedGraph<Weight>::GetIncidentEdges(VertexId vertex) const {
+	const auto &edges = incidence_lists_[vertex];
+	return {std::begin(edges), std::end(edges)};
+}
+}
+
+#endif // GRAPH_GRAPH_H_
```
Comment (foxfrikses) [inline lib/graph/src/graph.h:89]: Empty line.

File: lib/graph/src/router.h [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	return expanded_routes_cache_.at(route_id)[edge_idx];
+}
+
+template<typename Weight>
+void Router<Weight>::ReleaseRoute(RouteId route_id) {
+	expanded_routes_cache_.erase(route_id);
+}
+}
+
+#endif // GRAPH_ROUTE_H_
```
Comment (foxfrikses) [inline lib/graph/src/router.h:154]: Empty line.

File: lib/graph/CMakeLists.txt (hunk 1 above)
Comment (siarheishut) [reply lib/graph/CMakeLists.txt]: Done.

File: lib/graph/src/router.h (hunk 3 above)
Comment (siarheishut) [reply lib/graph/src/router.h]: Done.

File: lib/graph/src/graph.h (hunk 2 above)
Comment (siarheishut) [reply lib/graph/src/graph.h]: Done.

**Review by foxfrikses — APPROVED** [review]
LGTM

**Pull request merged**